src/pages/index.js

```diff
@@ -1,75 +1,193 @@
+
+      // Use toggleButtonState instead of manually disabling the button
+      formValidators['profile-form']._toggleButtonState();
+    })
+    .catch((err) => {
+      console.error("Error updating user profile:", err);
+    })
+    .finally(() => {
+      popupWithFormProfile.renderLoading(false); 
+    });
+}
+
+// Handle add card form submission
+function handleAddCardFormSubmit(formData) {
+  return api.addCard({ // The function should return the Promise
+    name: formData.title,
+    link: formData.url
+  })
+  .then((newCard) => {
+    const cardElement = createCard(newCard);
+    cardSection.addItem(cardElement);
+    popupWithFormAddCard._form.reset(); // Clear the form fields only after submission
+    popupWithFormAddCard.close();
+    
+    // After resetting the form, update the button state
+    formValidators['card-form']._toggleButtonState(); // Ensure the button is disabled if the inputs are empty
+  })
+  .catch((err) => {
+    console.error("Error adding card:", err);
+  })
+  .finally(() => {
+    popupWithFormAddCard.renderLoading(false);
+  });
+}
+
+
+
+// Handle avatar form submission
+function handleAvatarFormSubmit(formData) {
+  popupWithFormAvatar.renderLoading(true); 
+  return api.updateUserAvatar(formData.avatar)
+    .then((updatedUserData) => {
+      userInfo.setUserInfo({
+        name: updatedUserData.name,
+        description: updatedUserData.about,
+        avatar: updatedUserData.avatar,
+      });
+      popupWithFormAvatar.close();
+      popupWithFormAvatar._form.reset();  // Clear the form fields only after submission
+
+      // Use toggleButtonState instead of manually disabling the button
+      formValidators['avatar-form']._toggleButtonState();
+    })
+    .catch((err) => {
+      console.error("Error updating avatar:", err);
+    })
+    .finally(() => {
+      popupWithFormAvatar.renderLoading(false);
+    });
+}
+
+
+// Handle card deletion
+function handleDeleteCard(cardId, cardElement) {
+  return api.deleteCard(cardId) 
+    .then(() => {
+      cardElement.remove();
+      popupWithConfirmation.close();
+    })
+    .catch((err) => {
+      console.error("Error deleting card:", err);
+    });
+}
+
+// Create card
+function createCard(data) {
+  const card = new Card(
+    data,
+    "#card-template",
+    () => popupWithImage.open(data),
+    (cardId, cardElement) => popupWithConfirmation.open(cardId, cardElement),
+    api
+  );
+  return card.getView();
+}
+
+// Event listeners
+profileEditButton.addEventListener("click", () => {
+  const userInfoData = userInfo.getUserInfo();
+  popupWithFormProfile.setInputValues(userInfoData);
+  formValidators['profile-form'].resetValidation();
+  popupWithFormProfile.open();
+});
+
+addNewCardButton.addEventListener("click", () => {
+  popupWithFormAddCard.open();
+});
+
+avatarEditButton.addEventListener("click", () => {
+  formValidators['avatar-form'].resetValidation();
```

Comment:
`  formValidators['avatar-form'].resetValidation();
`

It's not needeed on opening of you reset the form and toggle the button in `then`